projects/core/src/cms/occ/occ-cms-component.adapter.ts

```diff
@@ -40,26 +41,55 @@ export class OccCmsComponentAdapter implements CmsComponentAdapter {
     pageSize = ids.length,
     sort?: string
   ): Observable<CmsComponent[]> {
-    const requestParams = this.getComponentsRequestParams(
+    const componentIds: ComponentIds = { componentIds: ids };
+    this.componentIdsFlag = true;
+
+    let requestParams = this.getComponentsRequestParams(
+      componentIds,
       pageContext,
+      this.componentIdsFlag,
       currentPage,
       pageSize,
       sort
     );
 
-    const idList: IdList = { idList: ids };
-
     return this.http
-      .post<CmsComponentList>(
+      .get<CmsComponentList>(
         this.getComponentsEndpoint(requestParams, fields),
-        idList,
         {
           headers: this.headers,
         }
       )
       .pipe(
         pluck('component'),
-        this.converter.pipeableMany(CMS_COMPONENT_NORMALIZER)
+        this.converter.pipeableMany(CMS_COMPONENT_NORMALIZER),
+        catchError(error => {
+          if (error.status === 400) {
+            this.componentIdsFlag = false;
+
+            requestParams = this.getComponentsRequestParams(
+              componentIds,
+              pageContext,
+              this.componentIdsFlag,
+              currentPage,
+              pageSize,
+              sort
+            );
+
+            return this.http
+              .post<CmsComponentList>(
+                this.getComponentsEndpoint(requestParams, fields),
+                componentIds,
```

Comment:
'post' requires `ComponentIDListWsDTO`, which is `{idList: string[];}`. So, the `IdList` model you removed is for this purpose.